### ISSUE TITLE
SignalR 0.5.3 hub doesn’t accept POST method for send in long polling, cross domain mode.  

### ISSUE THREAD

**Issue by RussellPolitzky** [SignalR/SignalR#599]
SignalR 0.5.3 hub doesn’t accept POST method for send in long polling, cross domain mode.

I’m using IIS Express with the newly released SignalR 0.5.3. 

If I enable cross domain by using the following URL

``` javascript
        $.connection.hub.url = connectionHubUrl;
        $.connection.hub.start()
        ...
```

then `send()` fails with the following message in Chrome:  

XMLHttpRequest cannot load http://localhost:50945/signalr/send?transport=longPolling&connectionId=fe4da756-826f-43e0-831f-56b6c1f25c30. Origin http://localhost:50944 is not allowed by Access-Control-Allow-Origin.

CORS calls in respect of GETs (negotiate and connect) do work as expected though.  
Manually adding  CORS headers as per below, doesn’t solve the problem and causes GET methods to fail too:

``` xml
<customHeaders>
  <add name="Access-Control-Allow-Origin" value="*" />
  <add name="Access-Control-Allow-Methods" value="GET, POST, PUT, DELETE, OPTIONS" />
  <add name="Access-Control-Allow-Credentials" value="true" />
</customHeaders>
```

According to the release docs of 0.5.1, it’s not necessary to add the custom headers above because all SignalR hubs now support CORS by default.

It seems that the default CORS pre-flight response from the SignalR hubs may not be sending back the correct set of methods although I’ve not observed  this traffic personally.

If I turn on JSONP as follows:

``` javascript
$.connection.hub.start({ jsonp: true })
```

so that CORS isn’t used, then cross domain calls work as expected.

In summary, there appears to be a bug with regards the CORS implementation in SignalR hubs in respect of `HTTP POST` methods.
